pages/auditlog/setup.rst

```diff
@@ -0,0 +1,248 @@
+
+.. _auditlog-config-option-mongodb-cleanup-interval:
+
+auditlog_mongodb_cleanup_interval
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This configures the interval of the  background job that periodically deletes
+old audit log entries from the MongoDB database. You have to use values like
+``1h`` (1 hour) to configure the interval.
+
+The default value for this is ``1h``.
+
+Example::
+
+    auditlog_mongodb_cleanup_interval = 1h
+
+.. _auditlog-config-option-mongodb-collection:
+
+auditlog_mongodb_collection
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This configures the name of the MongoDB collection where the audit log plugin#
+stores the audit log entries.
+
+The default value for this is ``audit_log``.
+
+Example::
+
+    auditlog_mongodb_collection = audit_log
+
+Log4j2 Configuration Options
+----------------------------
+
+The optional log4j2 audit log appender has a few configuration options available.
+
+.. note:: To configure the log4j2 appender you have to edit the Graylog server configuration file **and** the log4j2.xml file for your setup!
+
+.. list-table:: Configuration Options
+    :header-rows: 1
+    :widths: 7 20
+
+    * - Name
+      - Description
+    * - ``auditlog_log4j_enabled``
+      - whether the log4j2 appender is enabled or not
+    * - ``auditlog_log4j_logger_name``
+      - log4j2 logger name
+    * - ``auditlog_log4j_marker_name``
+      - log4j2 marker name
+
+.. _auditlog-config-option-log4j2-enabled:
+
+auditlog_log4j_enabled
+^^^^^^^^^^^^^^^^^^^^^^
+
+The log4j2 audit log appender is disabled by default and can be enabled by
+setting this option to ``true``.
+
+The default value for this is ``false``.
+
+Example::
+
+    auditlog_log4j_enabled = true
+
+.. _auditlog-config-option-log4j2-logger-name:
+
+auditlog_log4j_logger_name
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This configures the log4j2 logger name of the audit log.
+
+The default value for this is ``gl-org.graylog.plugins.auditlog``.
+
+Example::
+
+    auditlog_log4j_logger_name = graylog-auditlog
+
+.. _auditlog-config-option-log4j2-marker-name:
+
+auditlog_log4j_marker_name
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This configures the `log4j2 marker name <https://logging.apache.org/log4j/2.0/manual/markers.html>`_
+for the audit log.
+
+The default value for this is ``AUDIT_LOG``.
+
+Example::
+
+    auditlog_log4j_marker_name = AUDIT_LOG
+
+Log4j2 Appender Configuration
+-----------------------------
+
+To write audit log entries into a file you have to enable the log4j2 appender
+in your Graylog configuration file **and** add some configuration to the
+``log4j2.xml`` file that is used by your server process.
+
+The ``log4j2.xml`` file location is dependent on your deployment method.
+
+
+.. list-table:: Config File Locations
+    :header-rows: 1
+    :widths: 7 20
+
+    * - Deployment
+      - Location
+    * - OS Packages (deb, rpm)
+      - ``/etc/graylog/server/log4j2.xml``
+    * - Appliance
+      - tbd
+    * - manual
+      - no default ``log4j2.xml`` file available
+
```

Comment:
we have the special site for default file locations - maybe just link to? http://docs.graylog.org/en/2.1/pages/configuration/file_location.html